jaq-interpret/src/val.rs

```diff
@@ -55,8 +55,8 @@ fn rc_unwrap_or_clone<T: Clone>(a: Rc<T>) -> T {
 
 impl Val {
     /// Construct a string value.
-    pub fn str(s: String) -> Self {
-        Self::Str(s.into())
+    pub fn str(s: impl Into<String>) -> Self {
+        Self::Str(Rc::new(s.into()))
```

Comment:
I'm sorry, but I do not like this change because it makes the string function different from all other similar functions like arr, obj and so on. This might be useful, but only if all functions are changed accordingly, which I would however not do as part of this PR. Let's keep it small.